src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable react/prop-types */
+import React, { useContext } from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, act } from '@testing-library/react';
+import useClickTracker from '.';
+import pidginData from './fixtureData/tori-51745682.json';
+import {
+  RequestContext,
+  RequestContextProvider,
+} from '#contexts/RequestContext';
+import {
+  ServiceContext,
+  ServiceContextProvider,
+} from '#contexts/ServiceContext';
+import { STORY_PAGE } from '#app/routes/utils/pageTypes';
+import * as params from '#containers/ATIAnalytics/params';
+import * as beacon from '#containers/ATIAnalytics/beacon';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const urlToObject = url => {
+  const { origin, pathname, searchParams } = new URL(url);
+
+  return {
+    origin,
+    pathname,
+    searchParams: Object.fromEntries(searchParams),
+  };
+};
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
+process.env.SIMORGH_ATI_BASE_URL = 'https://logws1363.ati-host.net?';
+
+const defaultProps = {
+  pageData: pidginData,
+  componentName: 'brand',
+};
+
+const WithContexts = ({ children }) => (
+  <RequestContextProvider
+    bbcOrigin="https://www.test.bbc.com"
+    pageType={STORY_PAGE}
+    isAmp={false}
+    service="pidgin"
+    pathname="/pidgin/tori-51745682"
+  >
+    <ServiceContextProvider service="pidgin">{children}</ServiceContextProvider>
+  </RequestContextProvider>
+);
+
+const TestComponentContainer = () => {
+  const clickRef = useClickTracker(defaultProps);
+  return (
+    <div>
+      <TestComponent ref={clickRef} />
+    </div>
+  );
+};
+
+const TestComponent = React.forwardRef((_, ref) => {
+  return (
+    <div id="test-component" ref={ref}>
+      <a href="https://www.test.bbc.com">Testing Link</a>
+      <button type="button" id="test-button">
+        Testing Button
+      </button>
+    </div>
+  );
+});
+
+// eslint-disable-next-line react/prop-types
+const wrapper = ({ children }) => <WithContexts>{children}</WithContexts>;
+
+describe('Click tracking', () => {
+  it('should return a ref', () => {
+    const { result } = renderHook(() => useClickTracker(defaultProps), {
+      wrapper,
+    });
+
+    const { container } = render(
+      <WithContexts>
+        <TestComponent ref={result.current} />
+      </WithContexts>,
+    );
+
+    expect(result.current.current).toBe(
+      container.querySelector('#test-component'),
+    );
+  });
+
+  it('should call buildATIClickParams with page data, requestContext, serviceContext', () => {
```

Comment:
I don't think I see much value in this unit test. We could easily refactor your custom hook so it replaces requestContext, serviceContext and buildATIClickParams with some other implementation to successfully make calls to ATI on click events. This would mean the hook works fine but this test would say it's broken so you'd have to then spend time refactoring the tests to pass again even though you didn't actually break the hook functionality. It doesn't really matter that these functions were called. A test like this is kind of like pouring concrete around the code and making it difficult to refactor. I'm only saying this because I think we probably will want to refactor the ATI functions in the future. This article around not testing implementation details is worth a read https://kentcdodds.com/blog/testing-implementation-details also I'm more than happy to explain on a call!

Reply:
I just thought of this https://github.com/bbc/simorgh/pull/9071#discussion_r621287011

If I renamed buildATIClickParams function and filename and then updated the naming in view tracker and click tracker, things would still work in both hooks but this test would be broken meaning I have the extra chore of updating the test.

Reply:
Ahhh that makes a lot of sense. I'll remove the test and give that article a read, thanks!